### PR TITLE
[WP-549] provide accessibility text for loading button 

### PR DESCRIPTION
**Pull request by mortenko**
improved code in index.css file

### CODE REVIEW

File: src/example.css [hunk 1]
```diff
@@ -26,3 +26,8 @@
   text-align: center;
   background: #fff;
 }
+
+.loader-content-hidden {
```
Comment (tomiberes) [inline src/example.css]: Use BEM

File: src/index.css [hunk 2]
```diff
@@ -44,6 +44,13 @@
   background: var(--loading--overlay__background, #fff);
 }
 
+.loader-content-hidden {
```
Comment (tomiberes) [inline src/index.css]: Same here, use BEM. Why are you specifying `z-index: -1` also in example? It should be either explicit feature or just looks related change in example.

File: src/index.js [hunk 3]
```diff
@@ -19,12 +20,15 @@ export default class Loading extends React.Component {
   render() {
     const extraClassNames = this.props.className ? this.props.className.split(/\s+/g) : [];
     const els = [];
+    if (this.props.loadContent) {
+      els.push(<span className="loader-content-hidden" key="read-context" role="status">{this.props.loadContent}
```
Comment (tomiberes) [inline src/index.js]: Keep the `key` prop convention that can be seen in other elements (same as `className`).

File: src/index.css (hunk 2 above)
Comment (varholak-peter) [inline src/index.css]: Please use BEM notation: http://getbem.com/naming/

File: src/index.js (hunk 3 above)
Comment (varholak-peter) [inline src/index.js]: as above

File: src/example.js [hunk 4]
```diff
@@ -1,16 +1,25 @@
 import Loading from './index';
 import React from 'react';
 
+
```
Comment (umbobabo) [inline src/example.js]: This space is useless

File: src/index.js [hunk 5]
```diff
@@ -9,6 +9,7 @@ export default class Loading extends React.Component {
       overlay: React.PropTypes.bool,
       loadingText: React.PropTypes.string,
       delay: React.PropTypes.number,
+      loadContent: React.PropTypes.string.isRequired,
```
Comment (umbobabo) [inline src/index.js]: We don't need another message, we already have the 'loadingText' prop that is for the message that we want to give to the users.

File: src/index.js [hunk 6]
```diff
@@ -19,12 +20,15 @@ export default class Loading extends React.Component {
   render() {
     const extraClassNames = this.props.className ? this.props.className.split(/\s+/g) : [];
     const els = [];
+    if (this.props.loadContent) {
```
Comment (umbobabo) [inline src/index.js]: This part seems to me redundant, why not apply the span here >> https://github.com/economist-components/component-loading/pull/99/files#diff-1fdf421c05c1140f6d71444ea2b27638R35 and the role too?

File: src/example.js [hunk 7]
```diff
@@ -1,16 +1,25 @@
...
 export default (
   <div>
     <p>Default with no overlay</p>
-    <Loading />
+    <button>
+      <a href="#" >
```
Comment (umbobabo) [inline src/example.js]: This file is for example only, creating a button here and an anchor tag is not useful in this case.

File: src/example.js [hunk 8]
```diff
@@ -1,16 +1,25 @@
...
+      <Loading loadContent="the browser is loading new context" />
+     </a>
+    </button>
+    <Loading loadContent="the browser is loading new context" />
     <br />
     <div className="overlay">
       Just a test that is overlayed by the loader
-      <Loading loadingText="We are loading something..." className="loading--overlay" />
+      <Loading loadingText="We are loading something..." className="loading--overlay"
+        loadContent="the browser is loading new context"
```
Comment (umbobabo) [inline src/example.js]: See comment below.

File: src/index.css [hunk 9]
```diff
@@ -44,6 +44,13 @@
   background: var(--loading--overlay__background, #fff);
 }
 
+.loading__content {
```
Comment (umbobabo) [inline src/index.css]: I don't see a reason for this, see content below.

File: src/index.js [hunk 10]
```diff
@@ -19,12 +20,15 @@ export default class Loading extends React.Component {
...
+    }
     if (this.props.overlay) {
       els.push(<div className="loading__overlay" key="loading__overlay"></div>);
     }
 
     els.push(<div className="loading__inner" key="loading__inner"></div>);
-
     if (this.props.loadingText) {
       els.push(
         <div className="loading__text" key="loading__text">
```
Comment (umbobabo) [inline src/index.js]: This should probably become the Anchor tag that is required in the ticket...

Comment (umbobabo) [inline src/index.js]: L35 instead should probably become the span that is required in the ticket.

File: src/index.js [hunk 11]
```diff
@@ -7,7 +7,7 @@ export default class Loading extends React.Component {
     return {
       className: React.PropTypes.string,
       overlay: React.PropTypes.bool,
-      loadingText: React.PropTypes.string,
+      loadingText: React.PropTypes.string.isRequired,
```
Comment (varholak-peter) [inline src/index.js]: I would prefer we didn't change this to ```isRequired``` as it breaks backwards compatibility. We should rather define "The browser is loading new content" as a default prop, which could be overwritten as needed.

**Comment by tomiberes** [comment]
The requested changes have been applied, thus merging.

**Pull request merged**